common/src/address.rs

```diff
@@ -386,6 +386,65 @@ impl JsonSchema for IpRange {
     }
 }
 
+/// We need a custom deserialize to get better errors on the various ways a
+/// range can be invalid
+impl<'de> Deserialize<'de> for IpRange {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct RawIpRange {
+            first: String,
+            last: String,
+        }
+
+        let raw = RawIpRange::deserialize(deserializer)?;
+
+        // Attempt to parse all the kinds
+        // TODO: this should prob be done by parsing as Ipv4Range and Ipv6Range
```

Comment:
I don't think so? That precludes you from intelligible errors when someone (probably me) mixes ipv4 and ipv6 here.